pkg/image/registry/image/strategy.go

```diff
@@ -44,6 +45,25 @@ func (imageStrategy) AllowCreateOnUpdate() bool {
 	return false
 }
 
+func (imageStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (imageStrategy) PrepareForUpdate(obj, old runtime.Object) {
```

Comment:
I thought this is actually a composition of [Manifest.Name](https://github.com/docker/distribution/blob/master/manifest/schema1/manifest.go#L35) and [Manifest.Tag](https://github.com/docker/distribution/blob/master/manifest/schema1/manifest.go#L38). I believe that separation of name from manifest is on distribution's TODO (https://github.com/docker/distribution/pull/62#issuecomment-73991387).